Content.Server/GameObjects/Components/Chemistry/SmokeComponent.cs

```diff
@@ -0,0 +1,210 @@
+                        return;
+                }
+                var newSmoke = Owner.EntityManager.SpawnEntity("smoke_cloud", snapGrid.DirectionToGrid(dir));
+                var smokeComponent = newSmoke.GetComponent<SmokeComponent>();
+
+                if (Owner.TryGetComponent(out SolutionContainerComponent contents))
+                {
+                    var solution = contents.Solution.Clone();
+                    smokeComponent.TryAddSolution(solution);
+                }
+                smokeComponent.Start(_inception,_amount-1,_lifetime-_timer-_spreadDelay, _spreadDelay, _averageLifetime);
+            }
+
+            SpreadToDir(Direction.North);
+            SpreadToDir(Direction.East);
+            SpreadToDir(Direction.South);
+            SpreadToDir(Direction.West);
+        }
+
+        private void ReactWithTileAndEntities()
+        {
+            if (!Owner.TryGetComponent(out SolutionContainerComponent contents))
+                return;
+
+            var mapGrid = _mapManager.GetGrid(Owner.Transform.GridID);
+            var tile = mapGrid.GetTileRef(Owner.Transform.Coordinates.ToVector2i(Owner.EntityManager, _mapManager));
+
+            var solutionFraction = 1 / Math.Floor(_averageLifetime / ExposeDelay);
+
+            // Reagents react with the tile under the smoke
+            foreach (var reagentQuantity in contents.ReagentList.ToArray())
+            {
+                if (reagentQuantity.Quantity == ReagentUnit.Zero) continue;
+                var reagent = _prototypeManager.Index<ReagentPrototype>(reagentQuantity.ReagentId);
+                reagent.ReactionTile(tile, reagentQuantity.Quantity * solutionFraction);
+            }
+
+            foreach (var entity in tile.GetEntitiesInTileFast())
+            {
+                // Reagents touch all entities on the tile
+                foreach (var reagentQuantity in contents.ReagentList.ToArray())
+                {
+                    if (reagentQuantity.Quantity == ReagentUnit.Zero) continue;
+                    var reagent = _prototypeManager.Index<ReagentPrototype>(reagentQuantity.ReagentId);
+                    reagent.ReactionEntity(entity, ReactionMethod.Touch, reagentQuantity.Quantity * solutionFraction);
+                }
+
+                // Reagents enter the bloodstream of all entities without internals.
+                if (!entity.TryGetComponent(out BloodstreamComponent bloodstream))
+                    continue;
+
+                if (entity.TryGetComponent(out InternalsComponent internals) &&
+                    internals.BreathToolEntity != null &&
+                    internals.GasTankEntity != null &&
+                    internals.BreathToolEntity.TryGetComponent(out BreathToolComponent breathTool) &&
+                    breathTool.IsFunctional &&
+                    internals.GasTankEntity.TryGetComponent(out GasTankComponent gasTank) &&
+                    gasTank.Air != null)
+                    continue;
```

Comment:
Add a property to InternalsComponent  that answers "Will this person breath in chems" instead of doing the checks in the SmokeComponent